tests/test_backends.py

```diff
@@ -49,33 +52,99 @@ def test_get_backend_auto_detection_success_gguf(
-    tmp_path: pathlib.Path,
+# this tests both cases where a valid and invalid safetensors model directory is supplied
+@pytest.mark.parametrize(
+    "safetensors_dir,expected",
+    [
+        (
+            "valid_safetensors_model_dir",
+            True,
+        ),
+        (
+            "invalid_safetensors_model_dir",
+            False,
+        ),
+    ],
+)
+def test_is_model_safetensors_valid(
+    safetensors_dir: str, expected: bool, testdata_path: pathlib.Path
 ):
-    with pytest.raises(ValueError) as exc_info:
-        backends.get(tmp_path, None)
-    assert "Cannot determine which backend to use" in str(exc_info.value)
+    test_json_dict = {"a": 1, "b": 2}
+    json_object = json.dumps(test_json_dict, indent=4)
+    safetensors_model_path = testdata_path / safetensors_dir
 
+    for file in ["config.json", "tokenizer.json", "tokenizer_config.json"]:
+        os.makedirs(os.path.dirname(safetensors_model_path / file), exist_ok=True)
+        with open(safetensors_model_path / file, "a+", encoding="UTF-8") as f:
+            f.write(json_object)
 
-# this test succeeds because the model_path is a directory and the platform is linux so vllm is
+    with open(
+        safetensors_model_path / "model.safetensors", "a+", encoding="UTF-8"
+    ) as f:
+        f.write("")
+    if expected:
+        with open(
+            safetensors_model_path / "tokenizer.model", "a+", encoding="UTF-8"
+        ) as f:
+            f.write("")
+
+    val = backends.is_model_safetensors(safetensors_model_path)
+    assert val == expected
+    shutil.rmtree(safetensors_model_path)
+
+
+# this test succeeds because the model_path is a valid safetensors directory and the platform is linux so vllm is
 # picked
 @patch("sys.platform", "linux")
-def test_get_backend_auto_detection_success_vllm_dir(tmp_path: pathlib.Path):
-    backend = backends.get(tmp_path, None)
-    assert backend == "vllm"
+@pytest.mark.parametrize(
+    "safetensors_dir,expected",
+    [
+        (
+            "valid_safetensors_model_dir",
+            "vllm",
+        ),
+    ],
+)
+def test_get_backend_auto_detection_success_vllm_dir(
+    safetensors_dir: str, expected: bool, testdata_path: pathlib.Path
+):
+    test_json_dict = {"a": 1, "b": 2}
+    json_object = json.dumps(test_json_dict, indent=4)
+    safetensors_model_path = testdata_path / safetensors_dir
+
+    for file in ["config.json", "tokenizer.json", "tokenizer_config.json"]:
+        os.makedirs(os.path.dirname(safetensors_model_path / file), exist_ok=True)
+        with open(safetensors_model_path / file, "a+", encoding="UTF-8") as f:
+            f.write(json_object)
+
+    with open(
+        safetensors_model_path / "model.safetensors", "a+", encoding="UTF-8"
+    ) as f:
+        f.write("")
+
+    with open(safetensors_model_path / "tokenizer.model", "a+", encoding="UTF-8") as f:
+        f.write("")
+
+    backend = backends.get(safetensors_model_path, None)
+    assert backend == expected
```

Comment:
Is expected here supposed to be a `bool` or a `str`?  Isn't the value of backend here supposed to be "vllm"?

Reply:
it is "vllm", if you look at line 104

Reply:
fixed, thanks